### DISCUSSION TITLE
embedding php into redbean

### DISCUSSION THREAD

**Discussion by odarcan** [jart/cosmopolitan#1239] [General]
Hi I dont suppose there is a way of embedding a multi os php into redbean,
Im unsure this is the right place to discuss this

**Comment by alganet** [comment]
Hm, what exactly do you need from redbean?

If you want a PHP web server, you can download `lang.zip` from [superconfigure](https://github.com/ahgamut/superconfigure/releases), then:

```
$ ./php -S 0.0.0.0:8080
```

That will start the built-in PHP server. It has all you need.  You can even put stuff inside the ape and peek inside (eg `php  -r "print_r(glob('/zip/*'));"`), and you can use `.args` to setup a redbean-like server instrumented by php itself.

`php -S` is not a production server, redbean is much faster. However, it should do fine for hobby uses or behind a more capable reverse proxy. It should do extremely well if you plan to distribute it as a desktop app for a single user and not deploy it to a remote server.
